state/src/Collection.ts

```diff
@@ -99,6 +99,30 @@ export interface CollectionOptions<S, So, Si> {
   channel?: string;
 }
 
+interface HomogenousOptions<S, So, Si> extends BaseOptions<S, So, Si> {
+  /**
+   * The Cycle.js component for each item in this collection. Should be just a
+   * function from sources to sinks.
+   */
+  item(so: So): Si;
+
+  itemFactory?: never;
```

Comment:
This is clever, I like it!

Reply:
Yeah, this is a trick I figured out in manyverse https://github.com/staltz/manyverse/blob/c66f4ff059d8136414b8f503e2eb2806f7e75c36/src/frontend/screens/thread/props.ts#L20-L22 and since then have been using it sometimes in TypeScript. 

It's weird that you can't omit the field, it has to be there, and also it cannot be just `itemFactory: never`, it has to be `itemFactory?: never`. :shrug: